Source/Core/Core/IOS/DolphinDevice.h

```diff
@@ -14,5 +15,8 @@ class DolphinDevice final : public Device
   // Inherit the constructor from the Device class, since we don't need to do anything special.
   using Device::Device;
   IPCCommandResult IOCtlV(const IOCtlVRequest& request) override;
+
+private:
+  UICommon::GameFileCache cache;
```

Comment:
This will call the `GameFileCache` constructor that uses the default cache path, which works for DolphinQt but not for Android. For Android, you need to provide a specific path as an argument: https://github.com/dolphin-emu/dolphin/blob/9ef526c70888d91b27123e49218d7743aa5c3a0a/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/services/GameFileCacheService.java#L144

Reply:
Is there a way to do it for both DolphinQt and Android?

Reply:
I can't think of any solution to the problem that doesn't involve Android-specific code.

Reply:
Also, `getCacheDir()` is a Java API. Is there an equivalent for C++? I'd rather not use the UI's GameFileCache to prevent race conditions where a new game is added as the user is selecting a game.

Reply:
> Also, `getCacheDir()` is a Java API. Is there an equivalent for C++?

No, there is no C++ equivalent. You would need to call the Java function using JNI, or write a JNI function that can be used to pass the cache path from Java to C++.

> I'd rather not use the UI's GameFileCache to prevent race conditions where a new game is added as the user is selecting a game.

I agree, that doesn't sound like the way to go. It also wouldn't solve the Android problem, since you then would need separate code for accessing DolphinQt's GameFileCache and Android's GameFileCache.

Reply:
I don't really know anything about Android, and I don't have a toolchain for building the apk. Would it be okay if I just disabled this on Android for now?